### DISCUSSION TITLE
Support for User Groups

### DISCUSSION THREAD

**Discussion by sanmoh-hombal** [zitadel/zitadel#8400] [Q&A]
I've been fiddling around with Zitadel in a project and noticed that there's no support for grouping users. Few of the main use cases we want to use this feature for are as follows -

1. To map SSO Groups to the user groups in Zitadel
2. To assign roles to a group
3. To sync from LDAP user groups
4. Finally, in general all group CRUD operations

Would be helpful if there's any reference to workarounds for this

**Comment by fforootd** [comment]
ATM you could use an action to map the groups claim to either metadata or to a user grant.

For more conveniece we have these issue fo the topic https://github.com/zitadel/zitadel/issues/5822  https://github.com/zitadel/zitadel/issues/6270

**Reply by fforootd** [reply]
For some inspiration with actions v1 https://github.com/zitadel/actions/blob/main/examples/custom_roles_org_metadata.js
